language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/sourcegen/SourceGenTest.java

```diff
@@ -0,0 +1,138 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.sourcegen;
+
+import com.google.gson.JsonObject;
+import org.ballerinalang.langserver.SourceGen;
+import org.ballerinalang.langserver.compiler.DocumentServiceKeys;
+import org.ballerinalang.langserver.compiler.LSServiceOperationContext;
+import org.ballerinalang.langserver.compiler.format.TextDocumentFormatUtil;
+import org.ballerinalang.langserver.compiler.workspace.WorkspaceDocumentManager;
+import org.ballerinalang.langserver.compiler.workspace.WorkspaceDocumentManagerImpl;
+import org.ballerinalang.langserver.completion.util.FileUtils;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.FileVisitResult.SKIP_SUBTREE;
+
+/**
+ * Source generation test suit.
+ */
+public class SourceGenTest {
+    private Path examples = FileUtils.RES_DIR.resolve("sourceGen");
+    private List<File> ballerinaFiles;
+
+    @BeforeClass
+    public void loadExampleFiles() throws IOException {
+        this.ballerinaFiles = getExampleFiles();
+    }
+
+    @Test(description = "Source gen test", dataProvider = "exampleFiles")
+    public void sourceGenTests(File file) throws IOException {
+        LSServiceOperationContext formatContext = new LSServiceOperationContext();
+        Path filePath = Paths.get(file.getPath());
+        formatContext.put(DocumentServiceKeys.FILE_URI_KEY, filePath.toUri().toString());
+
+        WorkspaceDocumentManager documentManager = WorkspaceDocumentManagerImpl.getInstance();
+        byte[] encoded1 = Files.readAllBytes(filePath);
+        String expected = new String(encoded1);
+        documentManager.openFile(filePath, expected);
+        JsonObject ast = null;
+        try {
+            ast = TextDocumentFormatUtil.getAST(filePath.toUri().toString(), documentManager,
+                    formatContext);
+            SourceGen sourceGen = new SourceGen(0);
+            sourceGen.build(ast.getAsJsonObject("model"), null, "CompilationUnit");
+            String actual = sourceGen.getSourceOf(ast.getAsJsonObject("model"), false, false);
+            Assert.assertEquals(actual, expected, "Error in generated source");
+        } catch (InvocationTargetException | IllegalAccessException e) {
```

Comment:
Shall we throw these two exceptions as well?

Reply:
It is a file reading exception and it is not a part of this test. that's why we ignored this tests. But throwing them also is fine. I'll fix the PR.